### PR TITLE
Allow uchiwa to use chef-vault

### LINKED ISSUE #60: Add support for Chef-Vault
Need support for chef-vault for storing uchiwa credentials.

### PR DESCRIPTION
**Pull request by rajiv-g**
### Description

Allow uchiwa to use chef-vault.

### Issues Resolved

Fix #60 

### Check List

- [ ] All tests pass. See <https://github.com/chef-cookbooks/community_cookbook_documentation/blob/master/TESTING.MD>
- [ ] New functionality includes testing.
- [ ] New functionality has been documented in the README if applicable

### CODE REVIEW

File: libraries/uchiwa_helpers.rb [hunk 1]
```diff
@@ -1,9 +1,25 @@
 module Uchiwa
   module Helpers
 
+    extend ChefVaultCookbook if Kernel.const_defined?("ChefVaultCookbook")
     def self.data_bag_item(data_bag_name, data_bag_item, missing_ok=false)
 
-      Chef::EncryptedDataBagItem.load(data_bag_name, data_bag_item).to_hash.delete_if { |key, value| key == 'id' }
+      raw_hash = Chef::DataBagItem.load(data_bag_name, data_bag_item)
+      encrypted = raw_hash.detect do |key, value|
```
Comment (majormoses) [inline libraries/uchiwa_helpers.rb]: This seems like a much simpler way of doing this and comes from the readme so I think we should follow the same pattern: https://github.com/chef/chef-vault#determining-the-type-of-a-data-bag-item.

```
case ChefVault::Item.data_bag_item_type('data_bag_name', 'data_bag_item')
when :normal
  ...
when :encrypted
  ...
when :vault
  ...
end
```

**Comment by majormoses** [comment]
Thanks for contributing! without people like you we could not run the project. I will review shortly.

File: libraries/uchiwa_helpers.rb (hunk 1 above)
Comment (rajiv-g) [reply libraries/uchiwa_helpers.rb]: @majormoses Thanks for the suggestion. I have pushed changes.

**Comment by majormoses** [comment]
Thanks for that I will review this when I have some time.

**Comment by rajiv-g** [comment]
@majormoses Added change log entry as well as pushed changes as per your suggestion.

**Comment by majormoses** [comment]
Sorry this fell off my radar, reviewing.

**Comment by majormoses** [comment]
released: https://supermarket.chef.io/cookbooks/uchiwa/versions/1.5.0

**Pull request merged**